src/common.c

```diff
@@ -1373,9 +1373,9 @@ int stlink_load_device_params(stlink_t *sl) {
          sl->sram_size, sl->sram_size / 1024, sl->flash_size, sl->flash_size / 1024,
          (unsigned int)sl->flash_pgsz);
 #else
-    ILOG("%s: %zd KiB SRAM, %zd KiB flash in at least %zd %s pages.\n",
-         params->description, sl->sram_size / 1024, sl->flash_size / 1024,
-         (sl->flash_pgsz < 1024) ? sl->flash_pgsz  :  sl->flash_pgsz / 1024,
+    ILOG("%s: %u KiB SRAM, %u KiB flash in at least %u %s pages.\n",
+         params->description, (unsigned)(sl->sram_size / 1024), (unsigned)(sl->flash_size / 1024),
```

Comment:
@Nightwalker-87 Everything in this code is correct and there will be no errors on different configurations. The '%u' format corresponds to the 'unsigned' type.

Reply:
printf format-string know nothing about sized types. That provoke more issues using sized types in printf-like functions.  
uint32_t can be defined as unsigned int and as ling unsigned int. This leads us to use macro PRIu32, not just simple %u.

Reply:
@Ant-ON: I am talking about the use of typedefs like uint8, uint16, ... and int8, int16, ... instead of "native" int, unint and unsigned types. There is a reason for that, and this is a multi-platform toolset, so we should avoid bugs related to different interpretation of type sizes in different environments.

Reply:
@Nightwalker-87 Yes I agree. But it is more convenient to use cast to native types to display information. In my opinion macros like PRIu32 make it difficult to read the format string

Reply:
Whatsoever, let's proceed with this as it is now to get it through.
However this topic will likely be on the agenda again as soon as there is further refactoring / rewriting in future versions.
This implies that one should think about how to address this issue in the future.